### ISSUE TITLE
tagcloud.js - Contao-Combiner

### ISSUE THREAD

**Issue by marmeladenmopser** [hschottm/tags#70]
Hallo,

leider wird das Script tagcloud.js nicht über den Contao-Combiner mit anderen Scripts kombiniert, obwohl "Skripte zusammenfassen" im Seitenlayout aktiviert ist.
Gibts da ne einfache Lösung?

**Comment by hschottm** [comment]
Das mit der tags-Version ist erstmal korrekt. Die 4.5.x-Versionen sind für Contao 4.9.x, die 4.6.x-Versionen sind nur für Contao 4.10.x

Das tagcould-Skript muss ja von dir in deinen Frontend-Code eingebunden werden, das sollte doch wie bei allen anderen Javascript-Snippets problemlos möglich sein, wie hast du das Skript denn eingebunden?

**Comment by marmeladenmopser** [comment]
Hi,
super, daß du so schnell antwortest :)

Das wird von Contao automatisch eingebunden, sobald ein Modultyp "Tag Cloud" in die Seite eingebunden wird.
Hab jetzt auch schon herausgefunden, woran es liegt.
Das liegt an der Zeile 199 in modules/ModuleTagCloud.php:
`$GLOBALS['TL_JAVASCRIPT'][] = 'system/modules/tags/assets/tagcloud.js';`
Wenn ich das durch
`$GLOBALS['TL_JAVASCRIPT'][] = 'system/modules/tags/assets/tagcloud.js|static';`
ersetze, dann klappt es auch mit dem Script-Combinging.
Ich möchte aber wegen Updates nur ungern in deinem Code rumfummeln :(

**Comment by hschottm** [comment]
Ach, das ist das Javascript für die Tag-Eingabefelder im Backend.
Komisch, bei mir funktioniert das unter 4.9. Ich schau mir das mit dem static mal an, keine Ahnung wozu diese Option gut ist, wenn ich weiß, was es macht, ändere ich einfach die Integration.
Danke für den Hinweis auf jeden Fall schon mal.

**Comment by marmeladenmopser** [comment]
Ah, ok. Hab mir das Script gar nicht näher angeschaut, aber wenn es nur im Backend benötigt wird, dann wäre evtl.
`if (TL_MODE == 'BE') $GLOBALS['TL_JAVASCRIPT'][] = 'system/modules/tags/assets/tagcloud.js';`
die bessere Wahl.

**Comment by hschottm** [comment]
Nein, ist es nicht, ich bin jetzt schon total durcheinander. Das JavaScript wird für das TagCloud-Frontend-Modul eingebunden, es ist für die Top 10 und die Related Tags, die man per klick ausblenden kann.
Ich schaue mal, seit wann man dieses static mit angeben muss.

**Comment by hschottm** [comment]
Also, der static-Parameter soll eigentlich nur verwendet werden, wenn das Skript mehr oder weniger auf allen Contao-Seiten verwendet werden soll, weil es dann mit anderen Static-Skripten in ein kombiniertes Skript eingefügt wird. Ohne den Parameter wird das Skript als extra-Datei nur dann eingebunden, wenn es benötigt wird.
Da das Skript ja nur benötigt wird, wenn eine Tag Cloud verwendet wird, ist das eigentlich die korrekte vorgehensweise, static fügt das Skript auf allen anderen Seiten auch noch ein, obwohl es dort gar nicht benötigt wird.

**Comment by marmeladenmopser** [comment]
Ok, ich verstehe.
In meinem Fall ist es tatsächlich so, dass eine Tagcloud auf jeder Seite im Footer anzeigt wird.
Wäre es denkbar, bei der Modulkonfiguration eine Checkbox wie "auf allen Seiten einfügen" einzubauen, die dann den static-Parameter steuert?

Aber auch andersrum gedacht, wenn ich das Modul auf Seiten nicht verwende, also ein anderes Seitenlayout verwenden, dann würde es ja auch trotz "static" nicht eingebunden werden, oder?

**Comment by hschottm** [comment]
Das weiß ich eben nicht, so wie ich [die Dokumentation](https://docs.contao.org/dev/framework/asset-management/) lese, würde das Skript dann wohl immer eingebunden werden, selbst wenn das Modul gar nicht verwendet wird.

**Comment by marmeladenmopser** [comment]
Tatsächlich, würde ich das auch so verstehen wir du:
> A stylesheet should be defined as static, if it will occur on every or most pages of a website, ...

Aber ich habe das mal getestet, also den static-Parameter wieder angefügt.
Raus kam dabei im Quelltext:
`<script src="assets/js/megamenu.js,tagcloud.js-67d8c1dc.js"></script>`
Dann hab ich das Modul im Seitenlayout deaktiviert (bei verbleibendem static-Parameter) und es kam folgendes raus:
`<script src="assets/js/megamenu.js-69c0bd1a.js"></script>`
Daraus wäre zu folgern, dass es nicht auf allen Seiten eingebunden wird. Keine Ahnung wie obiges zu interpretieren ist.
